agent/Agent.js

```diff
@@ -431,6 +467,37 @@ class Agent extends EventEmitter {
     this.emit('updateProfileTimes', send);
   }
 
+  updateHooksTree(id: ElementID) {
+    const data = this.elementData.get(id);
+    const internalInstance = this.internalInstancesById.get(id);
+    if (internalInstance) {
+      const rendererID = this.rendererIdsByInternalInstance.get(internalInstance);
+      if (rendererID) {
+        const internals = this.reactInternals[rendererID].renderer;
+        if (internals && internals.currentDispatcherRef) {
+          // HACK: This leaks Fiber-specific logic into the Agent which is not ideal.
+          // $FlowFixMe
+          const currentFiber = data.state === internalInstance.memoizedState ? internalInstance : internalInstance.alternate;
```

Comment:
This is gross for obvious reasons, but it needs to live somewhere. Happy to move it out of the Agent if anyone thinks it's too gross even for a temporary solution.